### ISSUE TITLE
@context has a very special behaviour in jsonpath

### ISSUE THREAD

**Issue by vongohren** [decentralized-identity/presentation-exchange#278]
Is it worth mentioning anywhere that it needs escaping for certain languanges? Or is that common knowledge?

**Comment by JaceHensley** [comment]
Oh that's really interesting, it seems like the problem is that the original (? only one I could find) spec for JSON Path doesn't mention escaping at all

Related issues: https://github.com/dchester/jsonpath/issues/138, https://github.com/JSONPath-Plus/JSONPath/issues/154

**Comment by vongohren** [comment]
@JaceHensley yeah and as mentioned in the attached issue, there are certain comparission(standard) work going on. Link to github page working on this: https://github.com/cburgmer/json-path-comparison

**Comment by kimdhamilton** [comment]
I hate this bug

**Comment by nklomp** [comment]
LOL, this is a known issue in our PE implementation as well ;)

We hate it too @kimdhamilton , and have not gotten around to fixing it yet. 

The first reaction is to escape it on the PE implementation, but then there are also valid `@` possibilities that are not related to json-ld.  We probably will settle for having a list of known/mostly used JSON-LD keywords in our implementation.

**Comment by vongohren** [comment]
Glad to hear we share the pain, and I believe we will converge more on this going forward when more PE implementations surface. So believe this should stay open for a bit

**Comment by csuwildcat** [comment]
I played around with a bunch of JSONPath libs/evaluation sites and can't figure out if they all have done some augmentation or if this is already part of the grammar, but this appears to be supported:

Example object:
```json
{
 "@context": ["https://example.com"]
}
```

Expression:
```javascript
$..['@context']
```

**Comment by JaceHensley** [comment]
So the JSONPath string would just be `"$.['@context']"`

**Comment by vongohren** [comment]
@csuwildcat did you try out this test suite? https://github.com/cburgmer/json-path-comparison

**Comment by csuwildcat** [comment]
I filed an issue on the JSONPath IETF repo and they promptly informed me that querying for properties with special characters using bracket notation already is the standard form that allows for selection of properties that have characters like @ in them: https://github.com/ietf-wg-jsonpath/draft-ietf-jsonpath-base/issues/189

Any issues folks have brought up here must be with a lib that doesn't conform to the spec, but in my testing, most libs are handling this properly, as the spec directs them to.

**Comment by csuwildcat** [comment]
ACTION: not just Dan list the JSONPath libs that conform to the spec and allow you to select properties with `@` signs in them.

**Comment by gregsdennis** [comment]
> a lib that doesn't conform to the spec

Nothing confirms to the spec yet because the spec isn't done.  The entire reason we're writing a spec is that implementations deviate in their feature set.
